tests/gordo_components/server/test_base_view.py

```diff
@@ -8,42 +8,46 @@
 import numpy as np
 
 from gordo_components.server import utils as server_utils
-import tests.utils as tu
 
 
 @pytest.mark.parametrize(
-    "data_to_post",
+    "data_size,data_generation_method",
     [
-        np.random.random(size=(10, len(tu.SENSORS_STR_LIST))).tolist(),
-        np.random.random(size=(1, len(tu.SENSORS_STR_LIST))).tolist(),
-        pd.DataFrame(
-            np.random.random((10, len(tu.SENSORS_STR_LIST))),
-            columns=tu.SENSORS_STR_LIST,
-        ).to_dict("records"),
-        pd.DataFrame(
-            np.random.random((10, len(tu.SENSORS_STR_LIST))),
-            columns=tu.SENSORS_STR_LIST,
-        ).to_dict("list"),
-        pd.DataFrame(
-            np.random.random((10, len(tu.SENSORS_STR_LIST))),
-            columns=tu.SENSORS_STR_LIST,
-        ).to_dict("dict"),
+        (10, "np_list"),
+        (1, "np_list"),
+        (10, "dataframe_records"),
+        (10, "dataframe_list"),
+        (10, "dataframe_dict"),
     ],
 )
 @pytest.mark.parametrize("resp_format", ("json", "parquet", None))
 @pytest.mark.parametrize("send_as_parquet", (True, False))
 def test_prediction_endpoint_post_ok(
     base_route,
     sensors,
+    sensors_str,
     gordo_ml_server_client,
     data_to_post,
+    data_size,
+    data_generation_method,
     resp_format,
     send_as_parquet,
 ):
     """
     Test the expected successful data posts, by sending a variety of valid
     JSON formats of a dataframe, as well as parquet serializations.
     """
+    data_to_post = np.random.random(size=(data_size, len(sensors))).tolist()
+
+    if data_generation_method != "np_list":
+        df = pd.DataFrame(data_to_post, columns=sensors_str)
+        if data_generation_method == "dataframe_records":
+            data_to_post = df.to_dict("records")
+        elif data_generation_method == "dataframe_list":
+            data_to_post = df.to_dict("list")
+        elif data_generation_method == "dataframe_dict":
+            data_to_post = df.to_dict("dict")
+
```

Comment:
I'm for the change, but maybe just pass the actual argument for `to_dict(...)`, and have that variable be `None` if it the data should be left as a list.